### ISSUE TITLE
ui: Normalize CSS

### ISSUE THREAD

**Issue by peterzen** [decred/dcrdex#2708]
This issue is a lightweight spec for normalizing styles across the UI.  This is a change set that will touch almost all parts of the templates and the CSS, let's iterate and agree on an approach that's least likely to break things, results in a consistent look, is feasible to review and makes it easier to add GUI components with less CSS code going forward.

Goals:

* formalizing margins and padding
* consistent form and input and button styling
* get rid of the sans-light
* consolidating CSS

I'm suggesting this general approach:

* Replace CSS rules with Bootstrap classes wherever possible.  BS has classes for padding, margin, flex, font sizes expressed in `rem`, using these will improve the consistency of the layout and adapt better than `px` to different screen sizes.
* Replace hard coded color values with variables defined in `application.scss`
* Consolidate styles that are reused in different components into a utility CSS

Comments, suggestions are welcome and appreciated.

**Comment by buck54321** [comment]
Absolutely on board.

One area where I think we can affect a lot of normalization and consolidation is in table formatting. There should be maybe 2 to 4 table classes.

Another minor point to add is that we are inconsistent in the way we scope our page-specific classes. In e.g. *forms.scss*, we have
```SCSS
#regAssetForm {}
```
while in many other places we use
```SCSS
div[data-handler=markets] {}
```
I much prefer normalizing on the latter.

---

A couple of notes on the first point in the general approach.
> Replace CSS rules with Bootstrap classes wherever possible. BS has classes for padding, margin, flex, font sizes expressed in rem, using these will improve the consistency of the layout and adapt better than px to different screen sizes.

Using Bootstrap classes is and has always been a first approach. Probably close to 90% of the classes declared on elements are Bootstrap classes already. But I think you might be surprised at how many places it's not reasonable. In theory, we could use Bootstrap classes even in our own class definitions via `@extend`, but Bootstrap CSS often uses `!important` in places that I find unnecessary. 

Please don't touch the `.fs-` classes for now. `rem` units might be the right way to do it, but I need convincing. Maybe I just don't understand the techniques that make `rem` more suitable for responsive design. If so, please enlighten me. But browsers seem well adjusted to scaling `px` units in my experience, and `px` units are more intuitive.

**Comment by peterzen** [comment]
> 
> Using Bootstrap classes is and has always been a first approach. Probably close to 90% of the classes declared on elements are Bootstrap classes already. 

Indeed, there's actually not that many CSS rules that set spacing so that shouldn't be a big change.

> 
> Please don't touch the `.fs-` classes for now. `rem` units might be the right way to do it, but I need convincing. Maybe I just don't understand the techniques that make `rem` more suitable for responsive design. If so, please enlighten me. But browsers seem well adjusted to scaling `px` units in my experience, and `px` units are more intuitive.

Font size of `1 rem` equals to `16px`, this is set on the `<html>` element, other dimensions set in `rem` use a multiple of this so the spacing etc can adjust correctly to users' different default font settings.  Browsers are indeed not bad in scaling but the layout can look broken around the screen size breakpoints.

However, I think the accessibility aspect is really just a side effect - a more important feature of the `rem` based sizing is the consistent [vertical rhythm](https://imperavi.com/books/ui-typography/principles/vertical-rhythm/) we can get by using Bootstrap's spacing classes *together* with the typo sizing classes as these are set to round multiples and will look consistent and polished.  Font sizes are ought to be consolidated to the predefined `fs-*` classes anyway (no `font-size` in other rules unless it's absolutely necessary to make an exception) - the same consistency can be achieved in `px` units by using the same multiples but then why not just use `rem`.

**Comment by buck54321** [comment]
> However, I think the accessibility aspect is really just a side effect - a more important feature of the `rem` based sizing is the consistent [vertical rhythm](https://imperavi.com/books/ui-typography/principles/vertical-rhythm/) we can get by using Bootstrap's spacing classes _together_ with the typo sizing classes as these are set to round multiples and will look consistent and polished. Font sizes are ought to be consolidated to the predefined `fs-*` classes anyway (no `font-size` in other rules unless it's absolutely necessary to make an exception) - the same consistency can be achieved in `px` units by using the same multiples but then why not just use `rem` and get the accessibility benefit as well.

I"m only talking about font sizes. Don't touch the `.fs-` classes, please.
